mountpoint-s3/src/metrics/data.rs

```diff
@@ -147,15 +149,24 @@ impl AtomicGauge {
     }
 }
 
-/// A histogram with a hard-coded resolution that saturates at 300 seconds.
+/// A histogram with a hard-coded resolution that saturates at 300,000,000.
+///
+/// Histogram is usually used for time durations.
+/// As an example, this histogram can support up to 300 seconds for microsecond-precision measurements.
 #[derive(Debug)]
 pub struct Histogram {
     histogram: Mutex<hdrhistogram::Histogram<u64>>,
 }
 
 impl metrics::HistogramFn for Histogram {
     fn record(&self, value: f64) {
-        self.histogram.lock().unwrap().record(value as u64).unwrap();
+        let mut histogram = self.histogram.lock().unwrap();
+        match histogram.record(value as u64) {
+            Ok(_) => {}
+            Err(e) => {
+                error!("metric value {:?} could not be recorded: {:?}", value, e);
+            }
+        }
```

Comment:
Think we should probably just use `saturating_record` here instead -- 300s is long enough that we can pretty much interpret values in that bucket as "saturated".
```suggestion
        self.histogram.lock().unwrap().saturating_record(value as u64);
```

Reply:
Makes sense, although I'm a little worried that this is difficult to know as a customer. I'll see if there's any documentation that I can add a note to.

Reply:
I've instead enabled resizing and moved back to `record` with an `expect`.